src/components/fields/date-field/types.ts

```diff
@@ -11,11 +12,12 @@ export interface IDateFieldValidationRule extends IYupValidationRule {
 	minDate?: string | undefined;
 	maxDate?: string | undefined;
 	excludedDates?: string[] | undefined;
+	withinDays?: IWithinDaysRule;
```

Comment:
for all optional properties in FEE and DS, we apply `undefined` union. this is to cater to consumers who enabled `exactOptionalPropertyTypes` which will cause optional properties without the undefined union to be erroneous.

```suggestion
	withinDays?: IWithinDaysRule | undefined;
```

Reply:
I also notice the intention to use `IDateFieldSchema`'s `dateFormat` instead of the one from `withinDays`, I think this is fine but we will need to omit `dateFormat` prop from `IWithinDaysRule` here for clarity.